File: HDFC.PDFCoordinateMapper.WebV1/src/app/features/pdf-coordinate-mapper/amc-master/amc-master.models.ts
export type AmcMasterView = 'all' | 'approved';

export interface AmcMasterRecord {
  raw: Record<string, unknown>;
  autoId: string;
  mstColId: string;
  amcCode: string;
  amcName: string;
  amcDescription: string;
  active: string;
  status: string;
  action: string;
  actionRemark: string;
  createdBy: string;
  createdDate: string;
  modifiedBy: string;
  modifiedDate: string;
  approvedBy: string;
  approvedDate: string;
}

export interface AmcMasterFormValue {
  amcCode: string;
  amcName: string;
  amcDescription: string;
  active: string;
  remark: string;
}

export interface AmcMasterListSnapshot {
  amcs: AmcMasterRecord[];
  approvedAmcs: AmcMasterRecord[];
}

export interface AmcMasterDialogData {
  record: AmcMasterRecord | null;
}

File: HDFC.PDFCoordinateMapper.Api/Models/AmcMasterModels.cs
namespace HDFC.PDFCoordinateMapper.Api.Models
{
    public sealed class AmcMasterRequest
    {
        public string Flag { get; set; }
        public string Auto_Id { get; set; }
        public string Mst_Col_Id { get; set; }
        public string Amc_Code { get; set; }
        public string Amc_Name { get; set; }
        public string Amc_Description { get; set; }
        public string IsActive { get; set; }
        public string CurrentUserId { get; set; }
        public string Remark { get; set; }
    }
}
